libs/db/src/client.test.ts

```diff
@@ -134,25 +134,64 @@ test('transactions', () => {
     count: 1,
   });
 
-  expect(() =>
+  // Should roll back on exception:
+  await expect(
     client.transaction(() => {
       client.run('insert into muppets (name) values (?)', 'Fozzie');
       expect(client.one('select count(*) as count from muppets')).toEqual({
         count: 2,
       });
       throw new Error('rollback');
     })
-  ).toThrow('rollback');
+  ).rejects.toThrow('rollback');
   expect(client.one('select count(*) as count from muppets')).toEqual({
     count: 1,
   });
 
-  client.transaction(() => {
+  // Should commit by default, if no exceptions occur:
+  await client.transaction(() => {
     client.run('insert into muppets (name) values (?)', 'Fozzie');
   });
   expect(client.one('select count(*) as count from muppets')).toEqual({
     count: 2,
   });
+
+  // Should roll back if shouldCommit test returns false:
+  await expect(
+    client.transaction(
+      () => {
+        client.run('insert into muppets (name) values (?)', 'Gonzo');
+        expect(client.one('select count(*) as count from muppets')).toEqual({
+          count: 3,
+        });
+        return 'this is a result';
+      },
+      (result) => {
+        expect(result).toEqual('this is a result');
+        return false;
+      }
+    )
+  ).resolves.toEqual('this is a result');
+  expect(client.one('select count(*) as count from muppets')).toEqual({
+    count: 2,
+  });
+
+  // Should commit if shouldCommit test returns true:
+  await expect(
+    client.transaction(
+      async () => {
+        client.run('insert into muppets (name) values (?)', 'Gonzo');
+        return await Promise.resolve('another result');
+      },
```

Comment:
This makes odd use of `async`/`await` and `Promise.resolve`. I get that you're doing it because of the linter yelling at you, but maybe `client.transaction` should work with either sync or async callback functions? See [`mapSheet`](https://github.com/votingworks/vxsuite/blob/370fa214a1a41def2f01ee157cc28f8b9f130d16/services/scan/src/types.ts#L13-L41) for an example of this.

Reply:
was really only doing this in the test to verify that it works with async functions as well - will update `client.transaction` to the overloaded pattern for the other reasons pointed out, but this test  case might need to remain unchanged for coverage purposes

Reply:
Maybe just ditch the `async` and `await` and do `return Promise.resolve('another result');`?